test/include/util/test_thread_pool.h

```diff
@@ -34,61 +27,21 @@ class TestThreadPool {
    * @param repeat the number of times this should be done.
    */
   void RunThreadsUntilFinish(uint32_t num_threads, const std::function<void(uint32_t)> &workload, uint32_t repeat = 1) {
```

Comment:
Take in an thread pool and make this function a static util under some utility class